### PR TITLE
Conditional embedding cardano cli in docker image

### PR DESCRIPTION
**Pull request by sfauvel**
## Content
By default, build docker images (aggregator and signer) without Cardano CLI which is useless when using Pallas.

We can embed it setting to 1 the build arg EMBED-CARDANO-CLI.

Example:
`docker build -t mithril/mithril-aggregator --build-arg EMBED-CARDANO-CLI=1 -f mithril-aggregator/Dockerfile .`

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [x] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)
Closes 1725

### CODE REVIEW

File: mithril-aggregator/Dockerfile.ci [hunk 1]
```diff
@@ -29,7 +33,7 @@ COPY mithril-aggregator/config /app/config
 
 #Workdir
 WORKDIR /app/
-RUN chown -R appuser /app/ && chmod a+x /app/bin/mithril-aggregator && chmod a+x /app/bin/cardano-cli
+RUN chown -R appuser /app/ && chmod a+x /app/bin/mithril-aggregator
```
Comment (jpraynaud) [inline mithril-aggregator/Dockerfile.ci:36]: The chmod for the cardano-cli is not done anymore when `EMBED-CARDANO-CLI=1`

File: mithril-signer/Dockerfile.ci [hunk 2]
```diff
@@ -29,7 +34,7 @@ COPY mithril-signer/config /app/config
 
 # Workdir
 WORKDIR /app/
-RUN chown -R appuser /app/ && chmod a+x /app/bin/mithril-signer && chmod a+x /app/bin/cardano-cli
+RUN chown -R appuser /app/ && chmod a+x /app/bin/mithril-signer
```
Comment (jpraynaud) [inline mithril-signer/Dockerfile.ci:37]: The chmod for the cardano-cli is not done anymore when `EMBED-CARDANO-CLI=1`

Comment (sfauvel) [reply mithril-signer/Dockerfile.ci:37]: The cardano-cli comes from a downloaded archive which already have the execution rights.
This chmod seems to be needed when we download locally a signer from an artifact. In that case permissions may be lost (https://github.com/input-output-hk/mithril/commit/cdfa8780e9d5d242b85937b292bc62b2aecc0c13).
If you think it's safer to add the chmod command, I propose to add it at the end of the instruction into the conditional directive (just after the 'rm cardano-bin.tar.gz).
What is your opinion ?

**Review by Alenar — APPROVED** [review]

**Review by jpraynaud — APPROVED** [review]
LGTM

**Pull request merged**